docs/source/sysadmin_guide/pbft_adding_removing_node.rst

```diff
@@ -0,0 +1,158 @@
+   catch-up procedure, where it receives and commits the blocks that are already
+   on the blockchain. (For details, see the `PBFT node catch-up RFC
+   <https://github.com/hyperledger/sawtooth-rfcs/blob/master/text/0031-pbft-node-catchup.md>`_.)
+
+#. Wait for the node to catch up to the rest of the network (get within a few
+   blocks of the chain head). If the new node becomes a PBFT member before it
+   has caught up, the entire network could slow down or stall while the new
+   member node catches up.
+
+   This process can take a while, depending on the blockchain size, node's
+   hardware, and network speed. You can look at the logs or run
+   :ref:`sawtooth block list <sawtooth-block-list-label>` to check on the
+   new node's progress.
+
+#. Copy the new node's public validator key for use in the next step.
+
+   .. code-block:: console
+
+      $ cat /etc/sawtooth/keys/validator.pub
+
+   This step assumes that the validator key is stored in the default location,
+   ``/etc/sawtooth/keys``. If not, use the location specified by the ``key_dir``
+   setting (see :doc:`configuring_sawtooth/path_configuration_file`).
+
+#. On an existing member node, update ``sawtooth.consensus.pbft.members`` to
+   include the public validator key of the new node.
+
+   Run the following steps on a node that has permission to change on-chain
+   settings; that is, a node whose validator key is listed in
+   ``sawtooth.identity.allowed_keys``. Usually, the node that created the
+   genesis block is on this list. For more information, see
+   :ref:`config-onchain-txn-perm-label`.
+
+   a. List the current PBFT member nodes:
+
+      .. code-block:: console
+
+         $ sawtooth setting list --filter sawtooth.consensus.pbft.members
+
+   #. Submit a transaction that specifies the new list of all PBFT member nodes
+      (the previous list plus the new node's key).
+
+      .. Important::
+
+         BE VERY CAREFUL! Make sure to specify the full list of keys.
+         Double-check each key before you run this command, because a typo could
+         stall the network.
+
+      .. code-block:: console
+
+         $ sawset proposal create \
+           --key /etc/sawtooth/keys/validator.priv \
+           sawtooth.consensus.pbft.members=[previous-list, NEW-KEY]
+
+      If there are no errors, the block containing this setting change will be
+      committed quickly.
+
+#. When all nodes have detected the change and updated their local copy of the
+   member list, the new member node begins to participate in the PBFT network.
+
+.. _removing-a-pbft-node-label:
+
+Removing a PBFT Node
+====================
+
+To remove an existing node from a PBFT network, you will delete the node's
+validator key from  the ``sawtooth.consensus.pbft.members`` setting, then shut
+down the removed node. You can delete several nodes at the same time.
+
+.. Note::
+
+   PBFT consensus requires a network with at least four nodes. A network with
+   fewer than four nodes will fail.
```

Comment:
This is unfortunate (and unavoidable due to the algorithm) because there's also a low upper-bound of nodes (as the connections are O(n^2).

This implies to me that the genesis node must have at least four nodes. What happens if you configure, say, three nodes?

Reply:
That's correct; at genesis and any time thereafter, there need to be at least 4 nodes. If fewer than 4 nodes are configured in the `sawtooth.consensus.pbft.members` setting, PBFT will panic because it won't be able to operate.